packages/sdk/src/utils/pollApi.ts

```diff
@@ -34,7 +36,13 @@ export async function pollUntilHasData(
   // The response is still unchanged. Check again in five seconds
   await new Promise((resolve) => setTimeout(resolve, 5000))
   attemptCount++
-  await pollUntilHasData(request, dataParser, maximumAttempts, attemptCount)
+  await pollUntilHasData(
```

Comment:
Can we write a unit test to make sure that without data in the steps the executeSteps function will poll the api until data is found?